module/modmanager/manager.go

```diff
@@ -195,6 +197,37 @@ func (mgr *Manager) RemoveResource(ctx context.Context, name resource.Name) erro
 	return err
 }
 
+// ValidateConfig determines whether the given config is valid and returns its implicit
+// dependencies.
+func (mgr *Manager) ValidateConfig(ctx context.Context, cfg config.Component) ([]string, error) {
+	mgr.mu.RLock()
+	defer mgr.mu.RUnlock()
+	module, ok := mgr.getModule(cfg)
+	if !ok {
+		return nil,
+			errors.Errorf("no module registered to serve resource api %s and model %s",
+				cfg.ResourceName().Subtype, cfg.Model)
+	}
+
+	cfgProto, err := config.ComponentConfigToProto(&cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	// If no deadline is set on the context, add the default timeout.
+	if _, ok := ctx.Deadline(); !ok {
```

Comment:
I think it's okay and proffered to just add the timeout either way. It will end up being whichever timeout comes first anyway

Reply:
I don't think that's a case we're trying to support. The pattern we have in this codebase is to define a timeout specific to this operation

Reply:
Gotcha; removed the check for an existing deadline